### DISCUSSION TITLE
consider using `cargo fuzz`

### DISCUSSION THREAD

**Discussion by cosmicexplorer** [ethersync/ethersync#169] [General]
See https://github.com/ethersync/ethersync/pull/127#discussion_r1716701267. It might make sense to run our daemon and client through the fuzzing facilities provided by [`cargo fuzz`](https://rust-fuzz.github.io/book/cargo-fuzz.html), as a lower-level way to ensure safety. We can use [the `zip` crate](https://github.com/zip-rs/zip2/blob/master/fuzz/Cargo.toml#L8-L9) as an example.

**Comment by blinry** [comment]
I like the idea of "properly" fuzzing daemon components, but currently, I don't know what unit we could apply it to.

- Making completely arbitrary commands in Neovim?
- Making completely arbitrary random edits from the daemon?

On thing we've considered some time ago was fuzzing the OTServer component on its own, to make sure that is works correctly.

Until we have a better goal, I'm moving this PR to a discussion. If someone wants to pick this up, just let us know!
